migrations/20210913064936_addresses.ts

```diff
@@ -0,0 +1,65 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+  await knex.schema.createTable('addresses', (table) => {
+    table.uuid('id').primary().defaultTo(knex.raw('uuid_generate_v4()'));
+    table.uuid('psychologistId');
+    table.text('address');
+    table.string('departement');
+    table.string('region');
+    table.specificType('longitude', 'double precision');
+    table.specificType('latitude', 'double precision');
+  });
+
+  await knex.schema.table('addresses', (table) => {
+    table.foreign('psychologistId').references('psychologists.dossierNumber');
+    table.index('psychologistId');
+  });
+
+  const psychologists = await knex('psychologists');
+  const allAddresses = psychologists.flatMap((psychologist) => {
+    const addresses = psychologist.address.split(';');
+    return addresses.map((address) => ({
+      psychologistId: psychologist.dossierNumber,
+      departement: psychologist.departement,
+      region: psychologist.region,
+      address,
+    }));
+  });
+
+  if (allAddresses.length > 0) {
+    await knex('addresses').insert(allAddresses);
+  }
+
+  return knex.schema.table('psychologists', (table) => {
+    table.dropColumn('address');
+    table.dropColumn('region');
+    table.dropColumn('departement');
+    table.dropColumn('longitude');
+    table.dropColumn('latitude');
+  });
+}
+
+export async function down(knex: Knex): Promise<void> {
+  await knex.schema.table('psychologists', (table) => {
+    table.text('address');
+    table.string('departement');
+    table.string('region');
+    table.specificType('longitude', 'double precision');
+    table.specificType('latitude', 'double precision');
+  });
+
+  const addresses = await knex('addresses');
+
+  await Promise.all(addresses.map((address) => knex('psychologists')
+  .where({ dossierNumber: address.psychologistId })
+  .update({
+    departement: address.departement,
+    region: address.region,
+    address: address.address,
+    longitude: address.longitude,
+    latitude: address.latitude,
+  })));
+
+  return knex.schema.dropTable('addresses');
+}
```

Comment:
We can also remove the extension I added (finally not used because sorting is done on frontend side) : 
```
knex.raw(
    'DROP EXTENSION IF EXISTS "earthdistance"; \
     DROP EXTENSION IF EXISTS "cube";',
  )
 ```